joern-cli/frontends/php2cpg/src/main/scala/io/joern/php2cpg/passes/PhpTypeRecovery.scala

```diff
@@ -0,0 +1,448 @@
+    super.visitStatementsInBlock(b, assignmentTarget)
+  }
+
+  override protected def visitIdentifierAssignedToCall(i: Identifier, c: Call): Set[String] = {
+    logger.debug(s"visiting identifier ${i.name} assigned to call ${c.name}")
+    super.visitIdentifierAssignedToCall(i, c)
+  }
+
+  override protected def visitIdentifierAssignedToIdentifier(x: Identifier, y: Identifier): Set[String] = {
+    logger.debug(s"visiting identifier ${x.name} assigned to identifier ${y.name}")
+    super.visitIdentifierAssignedToIdentifier(x, y)
+  }
+
+  override protected def visitIdentifierAssignedToOperator(i: Identifier, c: Call, operation: String): Set[String] = {
+    logger.debug(s"visiting identifier ${i.name} assigned to operator ${operation}")
+    super.visitIdentifierAssignedToOperator(i, c, operation)
+  }
+
+  override protected def visitIdentifierAssignedToConstructor(i: Identifier, c: Call): Set[String] = {
+    logger.debug(s"visiting identifier ${i.name} assigned to constructor ${c.name}")
+    val constructorPaths = symbolTable.get(c).map(_.stripSuffix(s"${pathSep}<init>"))
+    associateTypes(i, constructorPaths)
+  }
+
+  override protected def visitIdentifierAssignedToCallRetVal(i: Identifier, c: Call): Set[String] = {
+    logger.debug(s"visiting identifier ${i.name} assigned to CallRetVal ${c.name}")
+
+    if (symbolTable.contains(c)) {
+      logger.debug(s"- symbol table contains call")
+      val callReturns = methodReturnValues(symbolTable.get(c).toSeq)
+      associateTypes(i, callReturns)
+    } else if (c.argument.exists(_.argumentIndex == 0)) {
+      logger.debug(s"- argument index exists")
+      val callFullNames = (c.argument(0) match {
+        case i: Identifier if symbolTable.contains(LocalVar(i.name))  => symbolTable.get(LocalVar(i.name))
+        case i: Identifier if symbolTable.contains(CallAlias(i.name)) => symbolTable.get(CallAlias(i.name))
+        case _                                                        => Set.empty
+      }).map(_.concat(s"$pathSep${c.name}")).toSeq
+      val callReturns = methodReturnValues(callFullNames)
+      associateTypes(i, callReturns)
+    } else {
+      // Check if the CPG already contains type info for this method, otherwise
+      // use dummy return value.
+      logger.debug(s"- checking CPG")
+      val rs = methodReturnValues(Seq(c.methodFullName))
+      if (rs.isEmpty) associateTypes(i, Set(s"${c.name}$pathSep${XTypeRecovery.DummyReturnType}"))
+      else associateTypes(i, rs)
+    }
+  }
+
+  override protected def visitIdentifierAssignedToLiteral(i: Identifier, l: Literal): Set[String] = {
+    logger.debug(s"visiting identifier ${i.name} assigned to literal ${l.code}")
+    super.visitIdentifierAssignedToLiteral(i, l)
+  }
+
+  override protected def visitIdentifierAssignedToMethodRef(
+    i: Identifier,
+    m: MethodRef,
+    rec: Option[String] = None
+  ): Set[String] = {
+    logger.debug(s"visiting identifier ${i.name} assigned to MethodRef ${m.code}")
```

Comment:
If the sole purpose for overriding these functions is to set debugging info, I'd rather you set the logging in the base class

Reply:
I'll remove these from my pass and consider whether what the appropriate amount of logging is in the base class. These were helpful for me getting started with learning how the type recovery system worked, but they do clutter up the logging messages (unless that's acceptable for the debug log level?).

Reply:
Right, likely more a thing for TRACE. But either way it looks like this is more-or-less the last thing left before this is ready I'd say.